libnmstate/nm/device.py

```diff
@@ -324,3 +324,13 @@ def get_device_common_info(dev):
         'type_name': dev.get_type_description(),
         'state': dev.get_state(),
     }
+
+
+def delete_device_inactive_profiles(ifname):
+    device_profiles = [
+        connection
+        for connection in nmclient.NM.Client.get_connections(nmclient.client())
+        if connection.get_interface_name() == ifname
+    ]
+    for profile in device_profiles:
+        profile.delete()
```

Comment:
The previous recommendation should fix this, but note that you cannot do this outside of the mainloop... and at this stage the mainloop has not yet been started.

Reply:
Now that should be fixed. Thanks!

Reply:
> list_connections_by_ifname should return in this case ConnectionProfile objects.

You have returned a NM object and not the nmstate one. Therefore the action of deletion is not performed in the mainloop.

Reply:
list_connections_by_ifname returns now ConnectionProfile objects. Thanks!